File: 4주차/42586. 기능개발/기능개발.java
import java.util.ArrayDeque;
import java.util.Queue;
import java.util.Arrays;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] answer = new int[progresses.length];
        
        Queue<Integer> qProgresses = new ArrayDeque<>();
        Queue<Integer> qSpeeds = new ArrayDeque<>();
        for(int i = 0; i < progresses.length; i++){
            qProgresses.add(progresses[i]);
            qSpeeds.add(speeds[i]);
        }
        int cnt = 0;
        int idx = 0;
        while(!qProgresses.isEmpty()){
            double p = qProgresses.poll();
            double s = qSpeeds.poll();
            
            if(p + s * cnt < 100)
                cnt = (int)Math.ceil((100.0 - p) / s);
            
            answer[idx] = cnt;
            idx++;
        }
        int[] answer2 = new int [(int)Arrays.stream(answer).distinct().count()];
        answer2[0] = 1;
        idx = 0;
        for(int i = 1; i < answer.length; i++){
            if(answer[i - 1] == answer[i]){
                answer2[idx] ++;
            }
            else{
                answer2[++idx] = 1;
            }
        }
        
        return answer2;
    }
}

File: 4주차/42586. 기능개발/기능개발.py
import queue
from re import A
def solution(progresses, speeds):
    answer = []
    q_program=queue.Queue()
    q_speeds=queue.Queue()
    
    for pro in progresses:
        q_program.put(pro)

    for sp in speeds:
        q_speeds.put(sp)

    day=0
    while q_program.qsize()!=0:
        pre_day=day
        program=q_program.get()
        speed=q_speeds.get()

        while True:
            if program+speed*day>99:
                break

            day=day+1

        if pre_day==day:
            answer[-1]=answer[-1]+1

        else:
            answer.append(1)

    return answer